pallets/core_mods/src/modules/did/base/mod.rs

```diff
@@ -91,3 +94,27 @@ impl<T: Config + Debug> Module<T> {
         Dids::<T>::insert(did, did_details.into())
     }
 }
+
+impl<T: Config> scale_info::TypeInfo for StoredDidDetails<T> {
```

Comment:
Why would derive `scale_info::TypeInfo` won't work?  Because of `VerRelType`?

Reply:
Because there's no other way to omit the type name prefix constructing from the file path.

Reply:
Which "type name prefix"? Where do we construct it from file path?

Reply:
Metadata V14 type name - it will be prefixed with the module path, see https://github.com/paritytech/scale-info/blob/master/derive/src/lib.rs#L124. 
We discussed it before but always can rename `Did` to `CoreModsModulesDidBaseDid` in `SDK`. 
As I already mentioned in our chat, a more appropriate solution would be to reorganize modules and define all prelude types in some root crate and build a correct tree structure. For ex., rename types like `Did` -> `DockDid`, `Controller` -> `DockDidController` etc.